src/RequestProcessor.cs

```diff
@@ -281,6 +337,151 @@ private void BindOutputFromResult(InvocationResponse response, AzFunctionInfo fu
+                if (Directory.Exists(moduleVersionFolderPath))
+                {
+                    // The latest version is already installed, log and return.
+                    var logMsg = string.Format(PowerShellWorkerStrings.ModuleIsAlreadyInstalled, moduleName, latestVersion);
+                    logger.Log(LogLevel.Trace, logMsg, isUserLog: true);
+
+                    return;
+                }
+
+                // Save-Module is able to download versions of Az module side-by-size. However, the dependent modules get
+                // overwritten with the latest version. As a result, the previous Az version will not longer work.
+                // If this is the case, empty the Az directory before running Save-Module.
+                if (moduleName.Equals(AzModuleName, StringComparison.OrdinalIgnoreCase))
+                {
+                    var azModulePath = Path.Join(path, moduleName);
+                    DependencyManagementUtils.EmptyDirectory(azModulePath);
+                }
+
+                // Save the module to the given path
+                SaveModuleCommand(pwsh, logger, moduleName, latestVersion, path);
+            }
+        }
+
+        /// <summary>
+        /// Downloads a PowerShell module from the PSGallery to the given path.
+        /// </summary>
+        internal void SaveModuleCommand(
+            System.Management.Automation.PowerShell pwsh,
+            ILogger logger,
+            string moduleName,
+            string version,
+            string path)
+        {
+            Exception exception = null;
+
+            // If the destination path does not exist, create it.
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            try
+            {
+                pwsh.AddCommand("PowerShellGet\\Save-Module")
+                    .AddParameter("Repository", Repository)
+                    .AddParameter("Name", moduleName)
+                    .AddParameter("RequiredVersion", version)
+                    .AddParameter("path", path)
+                    .AddParameter("Force", true)
+                    .AddParameter("ErrorAction", "Stop")
+                    .InvokeAndClearCommands();
+            }
+            catch (Exception e)
+            {
+                exception = e;
+                throw;
+            }
+            finally
+            {
+                if (pwsh.HadErrors)
+                {
+                    string errorMsg = string.Format(PowerShellWorkerStrings.FailToInstallModule, moduleName, version);
+                    logger.Log(LogLevel.Error, errorMsg, exception, isUserLog: true);
+                }
+                else
+                {
+                    var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, version);
+                    logger.Log(LogLevel.Trace, message, isUserLog: true);
```

Comment:
better to move this to the end of try block, so it's done when `pwsh.invoke` is successful.

Reply:
Really good point. Fixed.